.github/workflows/go.yml

```diff
@@ -0,0 +1,60 @@
+name: Go
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: "0 0 * * *"
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ["1.19","1.20", "1.x"]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Go ${{ matrix.go }}
+      uses: actions/setup-go@v2.1.3
+      with:
+        go-version: ${{ matrix.go }}
+    - name: Build
+      run: go build -v ./...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: '1.20'
+      - name: Run Coverage
```

Comment:
(nit) I find these a bit more readable with whitespace between each entry in `steps`, but this may be personal preference.